dotCMS/src/integration-test/java/com/dotmarketing/portlets/containers/business/FileAssetContainerUtilTest.java

```diff
@@ -9,6 +9,32 @@
 
 public class FileAssetContainerUtilTest extends ContentletBaseTest {
 
+    @Test
+    public void test_getPathFromFullPath_wrong_host_on_full_path() throws Exception {
+
+        final String relativePath = FileAssetContainerUtil.getInstance().getPathFromFullPath("demo.dotcms.com","//global.dotcms.com/application/container/test-container");
+
```

Comment:
I've never seen why FileAssetContainerUtil has to be a singleton. All the methods in there could be changed to static.
No state is kept inside of that class to justify making it a singleton. 
I think we should definitively make all those methods static and get rid of the getInstance.

Reply:
Another observation not directly related to this PR is that Optional is not meant to be used as parameters

Look at 

```
private void setContainerData(final Host host,
                                  final Folder containerFolder,
                                  final FileAsset metaInfoFileAsset,
                                  final List<FileAsset> containerStructures,
                                  final FileAssetContainer container,
                                  final Optional<String> preLoop,
                                  final Optional<String> postLoop,
                                  final Optional<FileAsset> preLoopAsset,
                                  final Optional<FileAsset> postLoopAsset,
                                  final String containerMetaInfo,
                                  final boolean includeHostOnPath,
                                  final Optional<String> codeScript)

```
https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments

Consider refactoring that, please.